compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -151,7 +151,7 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   case class CapturesAndResult(refs: List[Tree], parent: Tree)(implicit @constructorOnly src: SourceFile) extends TypTree
 
   /** Short-lived usage in typer, does not need copy/transform/fold infrastructure */
-  case class DependentTypeTree(tp: List[Symbol] => Type)(implicit @constructorOnly src: SourceFile) extends Tree
+  case class DependentTypeTree(tp: (List[TypeSymbol], List[TermSymbol]) => Type)(implicit @constructorOnly src: SourceFile) extends Tree
```

Comment:
Maybe we simplify this so that a DependentTypeTree just takes a `(tp: List[Symbol] => Type)`?

Reply:
Using a List[Symbol] means that we need to spend extra time concatenating and partitioning the parameters, and we need to document this behavior, so I'm not sure it's really more simple, but I'm happy to do the change if you prefer it that way.